### ISSUE TITLE
Node 12.8.0 and --h2 option leads to crash

### ISSUE THREAD

**Issue by petermuessig** [SAP/ui5-server#224]
## Expected Behavior
`ui5 serve --h2` works without error with Node 12.8.0.

## Current Behavior
When running `ui5 serve --h2` with Node 12.8.0 the Node application immediately crashes when trying to access resources from the server. The error is a  RangeError in some internals - not directly related to the UI5 tooling.

## Steps to reproduce the issue
1. Use Node 12.8.0
2. Run `ui5 serve --h2` on any UI5 project with any UI5 tooling version supporting h2

## Context
- UI5 Module Version *(output of `ui5 --version` when using the CLI)*: `1.4.3` but also happens with `1.7.0`
- Node.js Version: `12.8.0`
- npm Version: `6.10.3`
- OS/Platform: `MacOS`

## Affected components *(if known)*

- [ ] [ui5-builder](https://github.com/SAP/ui5-builder)
- [X] [ui5-server](https://github.com/SAP/ui5-server)
- [ ] [ui5-cli](https://github.com/SAP/ui5-cli)
- [ ] [ui5-fs](https://github.com/SAP/ui5-fs)
- [ ] [ui5-project](https://github.com/SAP/ui5-project)
- [ ] [ui5-logger](https://github.com/SAP/ui5-logger)

## Log Output / Stack Trace
```bash
URL: https://localhost:8443
internal/buffer.js:940
class FastBuffer extends Uint8Array {}
^

RangeError: Invalid typed array length: -4095
    at new Uint8Array (<anonymous>)
    at new FastBuffer (internal/buffer.js:940:1)
    at Handle.onStreamRead [as onread] (internal/stream_base_commons.js:165:17)
    at Stream.<anonymous> (/Users/d039071/Downloads/aoffline/node_modules/@ui5/cli/node_modules/handle-thing/lib/handle.js:120:12)
    at Stream.emit (events.js:208:15)
    at endReadableNT (/Users/d039071/Downloads/aoffline/node_modules/@ui5/cli/node_modules/readable-stream/lib/_stream_readable.js:1010:12)
    at processTicksAndRejections (internal/process/task_queues.js:77:11)
```

**Comment by petermuessig** [comment]
I opened the ticket here to document this behavior and to suggest the workaround to fallback to Node 10.x when you need H2 support of the UI5 server. With Node 10.x it works properly. 

If you are using homebrew you can switch back to an older version like this:
`brew unlink node && brew link node@10 --force --overwrite`

When switching back to Node 12.x you need to reinstall node, a simple unlink and link doesn't work here properly as npm, npx, ... are not re-linked properly. Use the following command then:
`brew unlink node@10 && brew reinstall node`

I don't know yet, whether we can fix this on UI5 tooling side. Needs investigation what goes wrong here.

**Comment by sebbi08** [comment]
Semms to be an issue with the `handle-thing` package they already got an issue for that https://github.com/spdy-http2/handle-thing/issues/6

**Comment by RandomByte** [comment]
This is a known issue with spdy.

Related issue: https://github.com/SAP/ui5-server/issues/77

Express will eventually support nodes native http/2 implementation in Express v5 (no eta): https://github.com/expressjs/express/pull/3730

I recently looked into other alternatives while working on the proxy PoC but couldn't a valid one.

As you said, workaround is to use Node 10. You can easily switch between Node versions using [nvm](https://github.com/nvm-sh/nvm).

**Comment by kristian** [comment]
+1 on this. Unfortunately Express has still not addressed this issue in their latest alpha release!

**Comment by RandomByte** [comment]
The upstream `spdy`/`handle-thing` issue has been fixed: https://github.com/spdy-http2/handle-thing/pull/13

Now the current master supports HTTP/2 with all current Node.js releases. For the UI5 CLI, this dependency update will be released with v2.0.0 next week.

This only requires an [in-range dependency update](https://github.com/SAP/ui5-server/commit/8a49066e162beee91ad33c529986b5d8a9d866f6) of handle-thing. @kristian your project should only need to update any package-lock.json to resolve this issue.

**Comment by kristian** [comment]
*cheer* Very nice! Thanks for the update @RandomByte!

**Comment by petermuessig** [comment]
👍 👍

**Comment by RandomByte** [comment]
Resolved in `@ui5/server@2.0.0`, part of `@ui5/cli@^2`

**Comment by petermuessig** [comment]
👍 Works, thanks!
